### PR TITLE
chore: handle unparsable URLs

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR makes it so that if we fail to parse an SSTP URL, instead of passing the mangled, incorrect URL to the App, we will instead send the original sstp URL.

## Why:
We have this thread https://returntocorp.slack.com/archives/C01NXGX2EHZ/p1674767808991039 in the slack where we were getting URLs that did not make sense. This was because the URLs that were read in from the environment variables were not parsable by our current implementation of `get_url_from_sstp_url`, and we were outputting a URL that contained `None` in it.

One proposed solution (discussed with @sabrinabrogren) was to port this URL-parsing logic to the App altogether, and just transmit the base URL. Unfortunately, there's several places in the code (such as here: https://github.com/returntocorp/semgrep/blob/22a0bd5b8aaae5a41a367902a9c5a2033db0c7c8/cli/src/semgrep/meta.py#L711) which rely on multiple environment variables, so I find this approach untenable, as the App would need to have access to all of them.

I believe that the next best option is just to avoid mangling the URL, if possible. That's what this PR does.

## How:
If we fail to parse any of the essential fields that are returned by the URL,

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by sabrinabrogren** [comment]
Ah, this looks like a reasonable approach. Nice! Would you mind adding a test?

File: cli/src/semgrep/meta.py [hunk 1]
```diff
@@ -40,6 +40,10 @@ def get_url_from_sstp_url(sstp_url: Optional[str]) -> Optional[str]:
         # let's just pick https
         protocol = "https"
 
+    # if we know that this URL is going to be unclickable, return the original URL
+    if None in [protocol, result.resource, result.owner, result.name]:
+        return sstp_url
```
Comment (aryx) [inline cli/src/semgrep/meta.py:50]: Can someone add a comment somewhere explaining what is an SSTP URL?
I never heard about SSTP.

Comment (brandonspark) [reply cli/src/semgrep/meta.py:50]: i don't really think i can write a comment about it, i've never heard of it either

all i know is that returning a URL which contains `None` in it is bad behavior

**Comment by brandonspark** [comment]
@sabrinabrogren I added a test, if you look at the `meta.json` file, you can see that the `repo_url` is correctly the original URL, and I've confirmed that without this change, it gets mangled.

File: cli/src/semgrep/meta.py (hunk 1 above)
Comment (sabrinabrogren) [reply cli/src/semgrep/meta.py:50]: ah yeah, perhaps we could add [this link](https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats) which talks about all the different git url formats.

**Review by sabrinabrogren — APPROVED** [review]

**Comment by aryx** [comment]
Nice link. Still don't know what is an sstp_url though :(

**Pull request merged**